java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java

```diff
@@ -103,8 +103,13 @@ public class JdkHttpClient implements HttpClient {
           new Authenticator() {
             @Override
             protected PasswordAuthentication getPasswordAuthentication() {
-              return new PasswordAuthentication(username, password.toCharArray());
+              if (username != null && password != null) {
```

Comment:
how about 

```
if (username == null || password == null) {
  return null;
}
return new PasswordAuthentication(username, password.toCharArray());
```

Reply:
I don't see this suggestion getting applied, @manuelsblanco.

Reply:
@diemol Sorry I was a couple of days out, getting applied now.

Reply:
@diemol The code was updated with the suggestions